`WellAnalysis` improvements

lindjoha opened the pull request:
Several improvements to the `WellAnalysis `plugin. Fixing some unwanted bahavoir I have seen when testing it on new datasets and some new features: well attribute filter and production after date.

---

### Contributor checklist

- [ ] :tada: This PR closes #ISSUE_NUMBER.
- [ ] :scroll: I have broken down my PR into the following tasks:
   - [x] New types PressurePlotMode, NodeType and ChartType
   - [x] Handling group trees that are different over realizations in well control plot
   - [x] Handling of interpolated WMCTL values (WMCTL is whole numbers in the Eclipse output, but can become decimal numbers in the presampled values.
   - [x] New functionality: Well attribute filter in the Well overview plots
   - [x] New functionality: Possible to see only the production after a given date in the Well overview plots
   - [x] Refactored the code in well_attributes_model.py and extended the tests
   
- [x] :robot: I have added tests, or extended existing tests, to cover any new features or bugs fixed in this PR.
- [x] :book: I have made an entry in `CHANGELOG.md`.

anders-kiaer reviewed (changes requested):
LGTM. 👍 🚀 Add a changelog entry and we are ready to 🚢?

Did @rnyb have some comments/ideas for future improvements? If so perhaps write it down as some short issues so we remember. 📔

lindjoha commented:
I have added a changelog entry and fixed two bugs that was discovered by @rnyb. The other comments are more open for discussion and not so much issues (yet). I will follow up on those.
